src/logic/equiv/basic.lean

```diff
@@ -817,6 +817,15 @@ product over `α` of `β (some α)` and `β none` -/
   left_inv := λ f, funext $ λ a, by cases a; refl,
   right_inv := λ x, by simp }
 
+/-- The dependant product of a type family `β : α → Type` is equivalent to the pair of `β a` and
+the dependant product of all but `a`, for any `a:α` -/
+@[simps] def pi_equiv_at_prod_not_at {α : Type*} [decidable_eq α] {β : α → Type*} (a : α) :
+   (Π x, β x) ≃ (β a) × (Π a', a' ≠ a → β a') :=
```

Comment:
I'm not sure why you switched from subtype to `Π a', a' ≠ a`? This seems harder to use.

Reply:
The following works:
```
@[simps] def pi_equiv_single_prod_erase {α : Type*} [decidable_eq α] (i : α) (β : α → Type*) :
  (Π j, β j) ≃ β i × Π j : {j // j ≠ i}, β j :=
{ to_fun := λ f, ⟨f i, λ j, f j⟩,
  inv_fun := λ f j, if h : j = i then h.symm.rec f.1 else f.2 ⟨j, h⟩,
  right_inv := λ f, by { ext, exacts [dif_pos rfl, (dif_neg x.2).trans (by cases x; refl)] },
  left_inv := λ f, by { ext, dsimp only, split_ifs, { subst h }, { refl } } }
```
You can make a shorter definition using existing constructions:
```
(equiv.pi_equiv_pi_subtype_prod _ β).trans $
  (equiv.Pi_subsingleton (λ j : {j // j = i}, β j) ⟨i, rfl⟩).prod_congr $ equiv.refl _ 
```
and `@[simps]` would generate the same simp lemmas (as confirmed by `@[simps]`, but it slows down `@[simps]` from 600-700ms to ~3s in the homeomorph version, so I don't go for it.

I've opened PR #16210 with the first definition, and specializes it to get a non-dependent version. The `eq.rec` would be annoying, but the `@[simps]` automatically removes it in the generated `simp` lemma, so I think it won't cause any trouble.

I don't include `merge_split_self` in the PR because it's only used once and can be easily inlined.

Reply:
#16210 has been merged. I'm polishing the PR and will push shortly.